src/iocore/net/quic/QUICStreamManager.cc

```diff
@@ -30,7 +30,20 @@
 
 QUICStreamManager::QUICStreamManager(QUICContext *context, QUICApplicationMap *app_map) : _context(context), _app_map(app_map) {}
 
-QUICStreamManager::~QUICStreamManager() {}
+QUICStreamManager::~QUICStreamManager()
+{
+  // We attempt to remove any stream that's left on the list.
+  if (!stream_list.empty()) {
+    QUICStream *stream = stream_list.head;
+    QUICStream *next   = nullptr;
+    while (stream) {
+      next = stream->link.next;
+      stream_list.remove(stream);
+      delete stream;
+      stream = next;
+    }
```

Comment:
You could just do:
```
    QUICStream *streamp = stream_list.head;
    stream_list.clear();
    while (streamp) {
      QUICKStream *next = streamp->link.next;
      delete streamp;
      streamp = next;
    }
```

Reply:
thanks Walt.
why is it better than the other?

Reply:
Accessing items in the list after calling `clear` *looks* odd. I guess it works fine for our DLL, but it doesn't look like a common way (it can be problematic if we switch the container).

Reply:
We could put the clear after the while loop.  But I can't get my code to wash Damian car, the way he seems to want.  It's not a necessary change.

Reply:
It seem like this destructor is only called when the connection is closed.  What if there was a connection that lasted a long time, with many streams opened and closed?  Can we remove the stream object from the list at the same time that the stream is closed?

Reply:
I'd call this a safety net or the last resort. Streams are (supposed to be) removed from the list when they are closed cleanly. This cleanup should be necessary only if a connection is closed abruptly (e.g. connection errors, active timeout, etc).